drivers/sensor/icm42688/icm42688_emul.c

```diff
@@ -41,6 +41,20 @@ void icm42688_emul_get_reg(const struct emul *target, uint8_t reg_addr, uint8_t
 	memcpy(val, data->reg + reg_addr, count);
 }
 
+static void icm42688_emul_handle_write(const struct emul *target, uint8_t regn, uint8_t value)
+{
+	struct icm42688_emul_data *data = target->data;
+
+	switch (regn) {
+	case REG_DEVICE_CONFIG:
+		if (FIELD_GET(BIT_SOFT_RESET, value) == 1) {
+			/* Perform a soft reset */
+			data->reg[REG_INT_STATUS] |= BIT_INT_STATUS_RESET_DONE;
```

Comment:
Is there any internal state that needs resetting?

Reply:
Done